core/src/main/java/org/elasticsearch/common/blobstore/fs/FsBlobContainer.java

```diff
@@ -85,7 +85,10 @@ public FsBlobContainer(FsBlobStore blobStore, BlobPath blobPath, Path path) {
     @Override
     public void deleteBlob(String blobName) throws IOException {
         Path blobPath = path.resolve(blobName);
-        Files.deleteIfExists(blobPath);
+        if (!Files.deleteIfExists(blobPath)) {
+            // blobPath does not exist
+            throw new IOException("File [" + blobPath.toString() + "] does not exist");
+        }
```

Comment:
Instead of making up our own exception, why not use just use Files.delete? This will give you a better exception message. https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#delete(java.nio.file.Path)